interfaces/builtin/utils.go

```diff
@@ -71,14 +75,19 @@ func plugAppLabelExpr(plug *interfaces.Plug) string {
 }
 
 // Function to support creation of udev snippet
-func udevUsbDeviceSnippet(subsystem string, usbVendor int64, usbProduct int64, key string, data string) string {
+func udevUsbDeviceSnippet(subsystem string, usbVendor int64, usbProduct int64, usbIterfaceNumber int64, key string, data string) string {
 	const udevHeader string = `IMPORT{builtin}="usb_id"`
-	const udevDevicePrefix string = `SUBSYSTEM=="%s", SUBSYSTEMS=="usb", ATTRS{idVendor}=="%04x", ATTRS{idProduct}=="%04x"`
+	const udevDevicePrefix string = `SUBSYSTEM=="%s", SUBSYSTEMS=="usb", ATTRS{idVendor}=="%04x", ATTRS{idProduct}=="%04x", ENV{ID_USB_INTERFACE_NUM}=="%02d"`
```

Comment:
The other numbers are hex, should we just use `%d` (and drop the leading decimal zero that looks confusingly like octal) or keep using `%04x` to match other values and be explicitly hexadecimal?

Reply:
I found the ENV{ID_USB_INTERFACE_NUM} is set by udev rule file `/lib/udev/rules.d/60-serial.rules`:
`SUBSYSTEMS=="usb", ENV{ID_USB_INTERFACE_NUM}="$attr{bInterfaceNumber}"`
So, the value ENV{ID_USB_INTERFACE_NUM} will be hex as well. I will use %02x instead.

It also means it would work for serial-port only, since it's from udev rules of serial device. I will remove those code in hidraw.